### ISSUE TITLE
Please add a setting to turn of animation on mouse wheel zoom

### ISSUE THREAD

**Issue by bencresty** [EsotericSoftware/spine-editor#676]
I'm using Spine for a long time, but working all day with Spine last weeks in a row now I see myself correcting the zoom on the viewport all the time, because the zoom animation often goes too far out when continuing after scrolling the wheel or too far in for the zoom level I aimed for. I tried real hard to get used to this behaviour, but after weeks of intensive usage in a row I know that this turned out to be an unachievable goal for me. 

The zoom animation looks great, but in practise it just doesn't work for me and is really getting in the way of the workflow now of me doing the work efficiently as I bump into this all the time.

There are settings to turn off UI animations and I did, but it doesn't turn off this one and I can't find another setting which does so assume it's not there.

Hoping you understand; could there please be added a setting to turn off these zoom-animations? That would be really helpful!

Thanks for considering!

**Comment by NathanSweet** [comment]
If you watch the zoom slider in the bottom left, the zoom level is set right away but the viewport zoom takes some time to get there. We can add a setting to reduce the time, but not in 4.1 because a new setting would require new text in the UI, which means a new round of translations. We try to only do translations at the end of a beta and to avoid adding untranslated text to non-beta versions.

I agree the current zoom duration is pretty long. We tried making it instant and that's a bit unpleasant. There are a number of settings that work together to define the zoom behavior. Whenever possible, if we can choose behavior that works for everyone, that would be better than adding a setting. We've made some improvements in 4.1.07, can you give that a try? Both scroll wheel and `U` are affected.

Please note that if your skeleton is very small or very large, that the `Bone scale` setting can help both to better size icon overlays in the viewport and to provide zoom behavior optimized for your skeleton size.

**Comment by bencresty** [comment]
@NathanSweet  Wow, just did a quick test and this zooming definitely feels a lot better now! Thanks! Will try it out next days.

One thing I noticed about zooming tho, hope you don't mind, is that zooming goes way faster when zoomed in compared to when zoomed out. It looks like zooming doesn't work in exponential steps, dependent on the current zoom level, but linear, making zooming feel 'slower' when zoomed out and faster sensitive when zoomed in. This is especially noticable when using the rotation touch wheel on the Wacom tablet which because of this behaviour is way more sensitive when zoomed in and makes even the slightest touch wheel rotation angle to zoom in or out very fast. The same seems to be the case when using the mouse wheel, although if feels easier to control with the mouse wheel. When working with Wacom I don't use the mouse tho.
Would be nice if zooming would feel like going the same speed independent on the current zoom level/scale. 

But is and leave that completely up to you of course. This is already much better! Thanks again!

**Comment by bencresty** [comment]
@NathanSweet That `Bone scale` setting is definitely very useful. Didn't know about that one before! Thanks for the tip!

**Comment by NathanSweet** [comment]
Do we still need to disable the zoom animation? It's pretty fast. If it doesn't bother I'd rather not have a setting. Too many settings gets overwhelming.

**Comment by bencresty** [comment]
@NathanSweet  Yes. Tbh I think animations in UI are quite disruptive and the reason for this request is still that. I am surprised you think Spine has that many settings and struggle to see why a checkbox extra would be 'overwhelming'. But if you really don't want to add a setting for that, than please remove all animations from the UI. It doesn't serve a functional purpose and makes using the interface harder.

**Comment by NathanSweet** [comment]
The reason Spine doesn't have too many settings currently is because we add them only when really required.

When have an `Interface animations` checkbox under `Behavior`. Unchecking that will remove the animation on zooming in 4.2.00-beta.

**Comment by bencresty** [comment]
I get where you're coming from and respect the amount of care and thought for quality you put into this. Turning the animation off with that existing setting sounds like a great and ideal solution to me! Kuddos for already implementing this too. That's awesome, thanks a lot!
